From 29403a7dd738879487628589b9fb3f90cf42cea6 Mon Sep 17 00:00:00 2001
From: "fei.deng" <fei.deng@amlogic.com>
Date: Wed, 15 Nov 2023 06:31:06 +0000
Subject: [PATCH 2/3] amlvideosink: CB2 send underflow msg [1/1]

PD#SWPL-144613

Problem:
NTS case failed because underflow msg lossed

Solution:
send underflow msg

Verify:
ah212

Change-Id: I1f4c4850c4bae653d337c3497193d585d64fae3f
Signed-off-by: fei.deng <fei.deng@amlogic.com>
---
 src/gstamlvideosink.c | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/src/gstamlvideosink.c b/src/gstamlvideosink.c
index 84156b0..45cc2c9 100644
--- a/src/gstamlvideosink.c
+++ b/src/gstamlvideosink.c
@@ -1437,8 +1437,11 @@ static void gst_render_msg_callback(void *userData, RenderMsgType type, void *ms
         g_signal_emit (G_OBJECT (sink), g_signals[SIGNAL_FIRSTFRAME], 0, 2, NULL);
     } break;
     case MSG_UNDER_FLOW: {
-        //GST_LOG_OBJECT(sink, "signal under flow");
-        //g_signal_emit (G_OBJECT (sink), g_signals[SIGNAL_UNDERFLOW], 0, 0, NULL);
+        GstAmlVideoSinkPrivate *sink_priv = GST_AML_VIDEO_SINK_GET_PRIVATE(sink);
+        if (!sink_priv->got_eos) {
+            GST_LOG_OBJECT(sink, "signal under flow");
+            g_signal_emit (G_OBJECT (sink), g_signals[SIGNAL_UNDERFLOW], 0, 0, NULL);
+        }
     } break;
     default:
     {
-- 
2.25.1

